### PR TITLE
Feature data dashboard

### PR DESCRIPTION
**Pull request by ketankartoza**
This PR is for Line Chart #659

Before :
Screenshot 2023-07-21 at 11 18 32 AM
After:
Screenshot 2023-07-21 at 11 20 19 AM

### CODE REVIEW

File: django_project/frontend/serializers/metrics.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+class SpeciesPopulationCountSerializer(serializers.ModelSerializer):
+    specie_name = serializers.SerializerMethodField()
+    specie_colour = serializers.SerializerMethodField()
+    annualpopulation_count = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Taxon
+        fields = [
+            'specie_name',
```
Comment (dimasciput) [inline django_project/frontend/serializers/metrics.py:16]: should be `species_name`

File: django_project/frontend/serializers/metrics.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            'specie_colour',
```
Comment (dimasciput) [inline django_project/frontend/serializers/metrics.py:17]: `species_colour`

File: django_project/frontend/serializers/metrics.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    def get_specie_name(self, obj):
+        return obj.common_name_varbatim
+
+    def get_specie_colour(self, obj):
+        return obj.colour
+
+    def get_annualpopulation_count(self, obj):
+        months = self.context['request'].GET.get('month')
+        owned_species = OwnedSpecies.objects.filter(taxon=obj)
+        annual_populations = AnnualPopulation.objects.filter(
```
Comment (dimasciput) [inline django_project/frontend/serializers/metrics.py:30]: @ketankartoza why not use single query?

``` 
annual_populations = AnnualPopulation.objects.filter(
        owned_species__taxon=obj,
        month__name__in=months.split(',') if months else F('month__name')
    ).values('month').annotate(month_total=Sum('total'))
```

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** FAILURE
**Build duration:** 7 min 55 sec

File: django_project/frontend/src/containers/MainPage/Metrics/SpeciesLineChart.tsx [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+const SpeciesLineChart = () => {
+    const selectedSpecies = useAppSelector((state: RootState) => state.SpeciesFilter.selectedSpecies)
+    const selectedMonths = useAppSelector((state: RootState) => state.SpeciesFilter.selectedMonths)
+    const startYear = useAppSelector((state: RootState) => state.SpeciesFilter.startYear)
+    const endYear = useAppSelector((state: RootState) => state.SpeciesFilter.endYear)
+    const [loading, setLoading] = useState(false)
+    const [selectedPropertyId, setSelectedPropertyId] = useState<number>(1)
+    const [propertyList, setPropertyList] = useState<PropertyInterface[]>([])
+    const [selectedProperty, setSelectedProperty] = useState<PropertyInterface>(propertyList?.[0])
```
Comment (dimasciput) [inline django_project/frontend/src/containers/MainPage/Metrics/SpeciesLineChart.tsx:37]: Why do you need both selectedPropertyId and selectedProperty? Could you derive selectedProperty from selectedPropertyId and propertyList?

**Review by dimasciput — CHANGES_REQUESTED** [review]

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** FAILURE
**Build duration:** 13 min

**Comment by ketankartoza** [comment]
@dimasciput I have done with the requested changes please review and merge

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** FAILURE
**Build duration:** 12 min

**Comment by dimasciput** [comment]
@ketankartoza please fix the lint errors

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** FAILURE
**Build duration:** 11 min

**Comment by ketankartoza** [comment]
> @ketankartoza please fix the lint errors

@dimasciput I have fixed the lint errors.

**Comment by dimasciput** [comment]
@ketankartoza please fix the broken test

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** FAILURE
**Build duration:** 11 min

**Comment by ketankartoza** [comment]
> @ketankartoza please fix the broken test



> @ketankartoza please fix the broken test

@dimasciput test case for this API is working fine, the previous test cases are failing.

**Comment by dimasciput** [comment]
@ketankartoza the build has failed : https://github.com/kartoza/sawps/actions/runs/5641209652/job/15278916527?pr=685. Could you please fix the issue?

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 15 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 17 min

**Review by dimasciput — APPROVED** [review]

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 14 min

**Pull request merged**